### PR TITLE
`scriptExecutor` should populate env values even when the script errors

### PR DESCRIPTION
**Pull request by m-Peter**
While working on Flow Emulator, trying to aggregate `computationUsed` and `memoryEstimate` metrics for scripts & transactions, I stumbled upon something that seems like a bug. When a script is reverted, e.g. due to exceeding computation limit, the `ProcedureOutput` is not properly populated.

**Before:** `computationUsed` & `memoryEstimate` are both `0`
reverted-script-before

**After:** `computationUsed` & `memoryEstimate` are both set
reverted-script-after

This is due to the fact that `scriptExecutor.executeScript()` returns early when it errors, and never calls `executor.output.PopulateEnvironmentValues(executor.env)`

**Note:** I can complement with a relevant test case, if you can point me to the most appropriate test file. I was not able to find a fitting place for this test case.

### CODE REVIEW

File: fvm/script.go [hunk 1]
```diff
@@ -198,11 +198,13 @@ func (executor *scriptExecutor) executeScript() error {
 			Source:    executor.proc.Script,
 			Arguments: executor.proc.Arguments,
 		},
-		common.ScriptLocation(executor.proc.ID))
+		common.ScriptLocation(executor.proc.ID),
+	)
+	populateErr := executor.output.PopulateEnvironmentValues(executor.env)
 	if err != nil {
 		return err
```
Comment (janezpodhostnik) [inline fvm/script.go:205]:
```suggestion
		return multierror.Append(err, populateErr)
```

so we don't accidentally loose the `populateErr` if both errors happen.

**Review by janezpodhostnik — CHANGES_REQUESTED** [review]

**Comment by janezpodhostnik** [comment]
Great find, thank you for the PR

**Comment by janezpodhostnik** [comment]
The tests for this would go in `fvm_test.go`. It would be slightly similar to `TestScriptAccountKeyMutationsFailure`

File: fvm/script.go (hunk 1 above)
Comment (m-Peter) [reply fvm/script.go:206]: Cool, didn't know about this :bow: 
Updated in https://github.com/onflow/flow-go/pull/4572/commits/574fb22f64c8dd2e60dc957bb3f8883f89195833

**Comment by m-Peter** [comment]
@janezpodhostnik Give me some time, and I will write some tests as well.

**Comment by m-Peter** [comment]
Tests added in https://github.com/onflow/flow-go/pull/4572/commits/1aa6e3dc006a989c623a429439a73a857e0ac140

**Review by koko1123 — APPROVED** [review]

**Comment by janezpodhostnik** [comment]
bors merge

**Pull request merged**